test/unit/initialize.test.js

```diff
@@ -65,4 +65,60 @@ describe('etch.initialize(component)', () => {
       etch.initialize(component)
     }).to.throw(/invalid falsy value/)
   })
+
+  it('allows for multi dimensional refs', async function () {
```

Comment:
You're right. Written a test for moving components around within refs and there were a couple of bugs with how `buildRefs.clear` works and with my edits to `updateRefs`.

I've fixed them now and added it to this PR.